test/server/drain_manager_impl_test.cc

```diff
@@ -37,12 +40,15 @@ TEST(DrainManagerImplTest, All) {
   // Test drain sequence.
   Event::MockTimer* drain_timer = new Event::MockTimer(&server.dispatcher_);
```

Comment:
This looks like it leaks memory.  Not a big deal for the test but can it be unique_ptr?